protocol-units/settlement/mcr/client/src/send_eth_tx.rs

```diff
@@ -0,0 +1,137 @@
+use crate::eth_client::McrEthConnectorError;
+use alloy_contract::CallBuilder;
+use alloy_contract::CallDecoder;
+use alloy_network::Ethereum;
+use alloy_provider::Provider;
+use alloy_transport::{Transport, TransportError};
+use std::marker::PhantomData;
+
+// Define a rule to verify the error generated when a tx is send to determine if:
+// * the Tx must me resend with more gas: return Ok(true)
+// * a specific error must be return: return Err(McrEthConnectorError::xxx);
+// * the rule doesn't apply: return Ok(false)
+pub trait VerifyRule: Sync + Send {
+	fn verify(&self, error: &alloy_contract::Error) -> Result<bool, McrEthConnectorError>;
+}
+
+pub struct SendTxErrorRule<Kind> {
+	_kind: PhantomData<Kind>,
+}
+
+impl<Kind> SendTxErrorRule<Kind> {
+	pub fn new() -> Self {
+		SendTxErrorRule { _kind: PhantomData::default() }
+	}
+}
+
+// Define the current 2 errors managed.
```

Comment:
Let's use https://docs.rs/thiserror/latest/thiserror as with #69 that will be predominant elsewhere in the workspace.